pysqldb3/pysqldb3.py

```diff
@@ -1228,15 +1228,37 @@ def xls_to_table(self, input_file=None, sheet_name=0, overwrite=False, schema=No
 
             if 'ogc_fid' in df.columns:
                 df = df.drop('ogc_fid', 1)
-            self.dataframe_to_table(df, table, schema=schema, overwrite=overwrite, temp=temp,
-                                    column_type_overrides=column_type_overrides, days=days)
+
+            if df.shape[0] > 100:
+                try:
+                    table_schema = self.dataframe_to_table_schema(df, table,
+                                                                  schema=schema,
+                                                                  overwrite=overwrite,
+                                                                  temp=temp,
+                                                                  column_type_overrides=column_type_overrides,
+                                                                  days=days)
+                    temp_file = os.path.dirname(
+                        input_file) + '\\'f'_temp_data_{datetime.datetime.now().strftime("%Y%m%d%H%M%S")}.csv'
+                    df.to_csv(temp_file, index=False, header=True)
+
+                    success = self._bulk_csv_to_table(input_file=temp_file, schema=schema, table=table,
+                                                      table_schema=table_schema, days=days)
+                    os.remove(temp_file)
+
+                    if not success:
+                        raise AssertionError('Bulk file loading failed.'.format(schema, table))
+
+                except SystemExit:
```

Comment:
if os.remove doesnt error if the file doesnt exist, could always try to remove in except as well if a problem happens midflight